### PR TITLE
Refactoring runtime conformance tests.

### LINKED ISSUE #55949: framework.ConformanceIt() fails with "called with non-literal argument"
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
Running conformance test and getting below error:
```
ERROR at test/e2e_node/runtime_conformance_test.go:316:29: framework.ConformanceIt() called with non-literal argument
```

ref: https://github.com/kubernetes/kubernetes/pull/54837#issuecomment-345091220

**What you expected to happen**:
The conformance test can be run successfully.

**How to reproduce it (as minimally and precisely as possible)**:
Below line is in a for loop.
https://github.com/kubernetes/kubernetes/blob/eff1a8463834c737da366b18cd7e65236e6a226e/test/e2e_node/runtime_conformance_test.go#L316

After changing it to `framework.ConformanceIt(testCase.description, func() {` and run `bazel test //test/conformance/...`, it throws the above error.

**Anything else we need to know?**:
If changing the above line to `framework.ConformanceIt("any literal string", func() {`, the tests pass.

/cc @mml 

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by yanxuean**
fix #55949
Signed-off-by: yanxuean <yan.xuean@zte.com.cn>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #55949 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/conformance/conformance_test.sh [hunk 1]
```diff
@@ -25,5 +25,5 @@ if diff -u test/conformance/testdata/conformance.txt test/conformance/conformanc
   echo PASS
   exit 0
 fi
-echo 'See instructions in test/conformance/README.md'
+echo 'See instructions in test/conformance/README.md'l
```
Comment (xiangpengzhao) [inline test/conformance/conformance_test.sh]: What's this line for?

File: test/e2e_node/runtime_conformance_test.go [hunk 2]
```diff
@@ -256,154 +256,176 @@ while true; do sleep 1; done
...
-					description:        "should be able to pull from private registry with credential provider",
+				"should be able to pull from private registry with credential provider": {
 					image:              "gcr.io/authenticated-image-pulling/alpine:3.1",
 					credentialProvider: true,
 					phase:              v1.PodRunning,
 					waiting:            false,
 				},
-			} {
-				testCase := testCase
-				It(testCase.description+" [Conformance]", func() {
```
Comment (dashpole) [inline test/e2e_node/runtime_conformance_test.go]: can we not just change this to framework.ConformanceIt()?

**Comment by dashpole** [comment]
Ahh, https://github.com/kubernetes/kubernetes/issues/55949#issuecomment-345316248

File: test/conformance/conformance_test.sh (hunk 1 above)
Comment (yanxuean) [reply test/conformance/conformance_test.sh]: My God, I don't know also. I have reset it.

**Comment by mml** [comment]
@yanxuean you need an approval from someone for test/e2e_node.  I can't grant that.

**Comment by yujuhong** [comment]
/approve

This does look very verbose, but I don't have any better suggestion.

**Comment by techtonik** [comment]
@yanxuean can you rebase this?

**Pull request closed without merging**